class/Defaults/MergeTag/Avatar.php

```diff
@@ -0,0 +1,76 @@
+<?php
+/**
+ * Avatar merge tag class
+ *
+ * Requirements:
+ * - Trigger property `user_object` or any other passed as
+ * `property_name` parameter. Must be an object, preferabely WP_User
+ *
+ * @package notification
+ */
+
+namespace BracketSpace\Notification\Defaults\MergeTag;
+
+use BracketSpace\Notification\Defaults\MergeTag\HtmlTag;
+
+/**
+ * Avatar merge tag class
+ */
+class AvatarTag extends HtmlTag {
+
+	/**
+	 * Trigger property to get the user data from
+	 *
+	 * @var string
+	 */
+	protected $property_name = 'user_object';
+
+	/**
+	 * Merge tag constructor
+	 *
+	 * @since [Next]
+	 * @param array $params merge tag configuration params.
+	 */
+	public function __construct( $params = [] ) {
+
+		if ( isset( $params['property_name'] ) && ! empty( $params['property_name'] ) ) {
+			$this->property_name = $params['property_name'];
+		}
+
+		$args = wp_parse_args(
+			$params,
+			[
+				'slug'        => 'avatar',
+				'name'        => __( 'User avatar', 'notification' ),
+				'description' => get_avatar( get_option( 'admin_email' ) ),
+				'example'     => true,
+				'resolver'    => function( $trigger ) {
+
+                    // check for 'avatar' property, fallback to 'user_email'
+					if ( isset( $trigger->{ $this->property_name }->avatar ) ) {
+                        return get_avatar( $trigger->{ $this->property_name }->avatar );
```

Comment:
Is there `avatar` property for `WP_User` object? Or this is from some 3rd party plugin?

Reply:
It should be functional for integration, if someone supplies an email address in `avatar` it ought to work, but's not actually used - when all was said and done, there was always a user object at hand, so it wasn't ever needed.  I can remove it.

Reply:
I think it will create more chaos than help, it immediately made me think if I'm missing something :) You could add a filter here to integrate with other services